### ISSUE TITLE
`VSCODE_PID` is not set inside extensions but available in Developer Console

### ISSUE THREAD

**Issue by gencer** [microsoft/vscode#98383]
- VSCode Version: **1.45.2**
- OS Version: **Win10 x64 2004**

Steps to Reproduce:

1. Install Ubuntu 20.04 LTS via WSL
2. Start VSCode from Windows via WSL
3. Extensions can't see `process.env.VSCODE_PID`

I need `VSCODE_PID`. This variable contains Windows Process ID instead of Ubuntu's node pid's. I exactly need this info. `Code.exe`'s PID.

Does this issue occur when all extensions are disabled?: **Yes**

I have an extension. I need to tell some apps my `Code.exe` PID.

This is what I have when I call `env` inside of an extension.

```
AMD_ENTRYPOINT: "vs/server/remoteExtensionHostProcess"
ANDROID_HOME: "/usr/lib/android-sdk"
APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL: "true"
CGO_LDFLAGS: ""
GOENV_ROOT: "/home/gencer/.goenv"
GOENV_SHELL: "bash"
GOPATH: "/home/gencer/go/1.14.2"
GOROOT: "/home/gencer/.goenv/versions/1.14.2"
HOME: "/home/gencer"
HOSTTYPE: "x86_64"
LANG: "C.UTF-8"
LOGNAME: "gencer"
LS_COLORS: "rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:"
NAME: "GENCER"
PATH: "/home/gencer/.vscode-server/bin/5763d909d5f12fe19f215cbfdd29a91c0fa9208a/bin:/home/gencer/.rbenv/bin:/home/gencer/.goenv/versions/1.14.2/bin:/home/gencer/.goenv/shims:/home/gencer/.goenv/bin:/home/gencer/.rbenv/shims:/home/gencer/.rbenv/bin:/home/gencer/.goenv/versions/1.14.2/bin:/home/gencer/.goenv/shims:/home/gencer/.goenv/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/mnt/c/Python38/Scripts/:/mnt/c/Python38/:/mnt/d/Ruby27-x64/msys64/mingw64/bin:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Windows/System32/OpenSSH/:/mnt/c/Users/gence/svg-resizer:/mnt/c/Qt/Qt5.14.2/5.14.2/msvc2017_64/bin:/mnt/d/go/bin:/mnt/c/Users/gence/svg-resizer/dll:/mnt/d/ASPACK/pgsql/bin:/mnt/d/bin:/mnt/d/Android/SDK/platform-tools:/mnt/d/nvm:/mnt/c/Program Files/nodejs:/mnt/c/Program Files/Amazon/AWSCLIV2/:/mnt/c/Program Files/CMake/bin:/mnt/c/Program Files/Git/cmd:/mnt/c/Go/bin:/mnt/c/Program Files/TortoiseGit/bin:/mnt/c/Program Files (x86)/WinSCP/:/mnt/c/Program Files/dotnet/:/mnt/c/Program Files/Microsoft SQL Server/130/Tools/Binn/:/mnt/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn/:/mnt/d/Ruby27-x64/bin:/mnt/c/Users/gence/AppData/Local/Microsoft/WindowsApps:/mnt/c/Program Files (x86)/GitHub CLI/:/mnt/c/Users/gence/AppData/Local/GitHubDesktop/bin:/mnt/c/Users/gence/go/bin:/mnt/c/Users/gence/AppData/Local/Programs/Microsoft VS Code/bin:/home/gencer/go/1.14.2/bin:/home/gencer/.bin:/snap/bin:/home/gencer/go/1.14.2/bin:/home/gencer/.bin:/usr/lib/android-sdk/tools/bin:/usr/lib/android-sdk/platform-tools"
PIPE_LOGGING: "true"
PWD: "/mnt/c/Users/gence/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.2"
RBENV_SHELL: "bash"
SHELL: "/bin/bash"
SHLVL: "0"
TERM: "xterm-256color"
USER: "gencer"
VERBOSE_LOGGING: "true"
VSCODE_AGENT_FOLDER: "/home/gencer/.vscode-server"
VSCODE_EXTHOST_WILL_SEND_SOCKET: "true"
VSCODE_HANDLES_UNCAUGHT_ERRORS: "true"
VSCODE_INJECT_NODE_MODULE_LOOKUP_PATH: "/home/gencer/.vscode-server/bin/5763d909d5f12fe19f215cbfdd29a91c0fa9208a/remote/node_modules"
VSCODE_IPC_HOOK_CLI: "/tmp/vscode-ipc-36a3d8b8-c31c-439a-ac3d-cc3e94936fe9.sock"
VSCODE_LOGS: "/home/gencer/.vscode-server/data/logs/20200522T165351"
VSCODE_LOG_STACK: "false"
VSCODE_NLS_CONFIG: "{"locale":"en","availableLanguages":{}}"
VSCODE_WSL_EXT_LOCATION: "/mnt/c/Users/gence/.vscode/extensions/ms-vscode-remote.remote-wsl-0.44.2"
WSLENV: "VSCODE_WSL_EXT_LOCATION/up"
WSL_DISTRO_NAME: "Ubuntu-20.04"
WSL_INTEROP: "/run/WSL/3395_interop"
XDG_DATA_DIRS: "/usr/local/share:/usr/share:/var/lib/snapd/desktop"
_: "/home/gencer/.vscode-server/bin/5763d909d5f12fe19f215
```

One more time i want to be clear on this. I don't want Ubuntu's Server `node` PID. I need to get `Code.exe`'s Windows PID from/in WSL.

When I toggle developer tools and write `process.env.VSCODE_PID`, I can see Code.exe's PID. Why not this available in extension? **How can I access to this information?**

**Comment by aeschli** [comment]
Can you explain what you want to do with that PID?

Typically client and remote extension hosts runs on a different platform and the PID of the client has no meaning in the remote environment. The exception is WSL1, but you wouldn't want to write an extension that only works in WSL1.

**Comment by gencer** [comment]
@aeschli , I am using Logitech Craft Keyboard and it has an extension in the gallery. Logitech Options is a program that controls keyboard actions. It works like this:

1. VSCode starts
2. Extension connects to `127.0.0.1:10134` (Options)
3. After successful connect, Extension reports VSCode PID and exeName. This means Options will send required keystrokes or actions to the exact same visual studio code instance if its opened multiple times. Otherwise it will be a mess.

Now, I tried on Remote SSH. VSCODE_PID doesn't set on there too. I need to let Logitech Options to know my Windows PID so locally it can send events to that process. 

I handled `IP:PORT` redirection on my behalf. Only thing that remained is to access my Windows' Visual Studio Code PID from SSH/WSL Extension.

See extension [here](https://github.com/idolize/logitech-craft-vscode). And see the related code [here](https://github.com/idolize/logitech-craft-plugin/blob/b1f02b4c1c166d819b07459a3c5cad2e06d1907c/src/CraftPlugin.ts#L10)
